ui/src/app/core/gcs.service.ts

```diff
@@ -0,0 +1,33 @@
+import {Injectable} from '@angular/core';
+import {AuthService} from '../core/auth.service';
+
+declare const gapi: any;
+
+/** Thin wrapper around the Google cloudresourcemanager API project.list(). */
+@Injectable()
+export class GcsService {
+
+  private static readonly apiPrefix = "https://www.googleapis.com/storage/v1"
+
+  constructor(private readonly authService: AuthService) {}
+
+  readObject(bucket: string, object: string): Promise<string> {
+    return this.authService.isAuthenticated().then( authenticated => {
+      if (authenticated) {
+        return gapi.client.request({
+          path: `${GcsService.apiPrefix}/b/${bucket}/o/${object}`,
+          params: {alt: 'media'}
+        })
+        .then(response => response.body)
+         // If an error occurs ignore it, the file may not exist
+        .catch(response => "")
```

Comment:
Good point. Can we detect that case specifically here? Would be nice to continue doing what you are now for "does not exist" (note, this might be a 403 rather than a 404, don't remember what GCS does) and give some notification or show some text here for "random transient error".